tests/tests_data/streaming/test_cache.py

```diff
@@ -242,3 +243,31 @@ def test_streaming_dataset(tmpdir, monkeypatch):
 
     dataloader = DataLoader(dataset, num_workers=2, batch_size=2)
     assert len(dataloader) == 408
+
+
+def test_create_oversized_chunk_single_item(tmp_path):
+    cache = Cache(str(tmp_path), chunk_bytes=700)
+    with pytest.warns(UserWarning, match="An item was larger than the target chunk size"):
+        cache[0] = np.random.randint(0, 10, size=(10000,), dtype=np.uint8)
+
+
+def test_create_undersized_and_oversized_chunk(tmp_path):
+    cache = Cache(str(tmp_path), chunk_bytes=9000)  # target: 9KB chunks
+    with no_warning_call(UserWarning):
+        cache[0] = np.random.randint(0, 10, size=(500,), dtype=np.uint8)
+        cache[1] = np.random.randint(
+            0, 10, size=(10000,), dtype=np.uint8
+        )  # this item won't fit in the target chunk size
+    with pytest.warns(UserWarning, match="An item was larger than the target chunk size"):
+        cache[2] = np.random.randint(0, 10, size=(150,), dtype=np.uint8)
+    with no_warning_call(UserWarning):
+        cache[3] = np.random.randint(0, 10, size=(200,), dtype=np.uint8)
+
+    cache.done()
+    cache.merge()
+    assert len(os.listdir(tmp_path)) == 4  # 3 chunks + 1 index file
+    metadata_bytes = 167
+    assert os.path.getsize(tmp_path / "chunk-0-0.bin") == 500 + metadata_bytes
+    assert os.path.getsize(tmp_path / "chunk-0-1.bin") == 10000 + metadata_bytes
+    # TODO: Why doesn't this assertion hold?
+    # assert os.path.getsize(tmp_path / "chunk-0-2.bin") == 150 + 200 + metadata_bytes
```

Comment:
@tchaton I have no idea why the file size isn't right here

Reply:
What's the size ?

Reply:
668
if we subtract these 167 bytes that the others have extra, we get 501 🤔

Reply:
I think it is the pickle overhead. We don't have a numpy serializer.

Reply:
I revised the test, asserting the number of items per chunk via the config file